### PR TITLE
Refactor height & Introduce species heights

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

Swaps our current height system, a series of transforms, to Byond's displacement filter.

Introduces species heights, variable height standards for each species. A 'Tall' Psyphoza will only be as tall as a 'Normal' human.

Features like dwarfism & gigantism have not been adjusted, as they work with the height system regardless of implementation.

For the next part, it's important you understand how the height values work. The values we pass into height represent the displacement size of the filter used to shorten, or lengthen, a mob. However, these values don't represent an exact amount of change. This means a height value of 8 won't decrease a mobs height by 8, but it *is* shorter than 0. Negative heights are tall, and positive heights are short.

The 'Tall' 'Normal' and 'Short' heights for humans map to -8, 0, 8 respectively. However, these values for mothmen, apids, flypeople, felinids, and psyphoza has been changed to 0, 8, 16. This helps us represent species that are typically shorter than humans, but have at-least one height setting that is 'normal'.

## Why It's Good For The Game

The current height system has many artifacts, and isn't exceptionally flexible or good-looking. This looks better and offers more options.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

image

image

image

image

</details>

## Changelog
:cl:
tweak: Make height code use displacement maps
add: Add variable heights to species
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
The short looks way too short imo, especially on the species that are 8 shorter than others.

**Comment by DrDuckedGoose** [comment]
> To prevent this from being gamed, could a very small movement nerf be applied to those who are shorter? It'd make sense compared to real life.

No can do big fella
Keep in mind we're an RP server with rules and administration to prevent powergaming. I don't especially like features that *force* you to play a certain way, especially for the pursuit of 'realism' or 'RP'.

**Comment by MarkusLarsson421** [comment]
> Keep in mind we're an RP server with rules and administration to prevent powergaming.

I'm no admin, but I imagine not many people has been bwoinked for trying to minimize their sprite size to get that small combat advantage. This feels like another subtle tool in their arsenal. Plus, I think this could benefit RP as then tall people could lightly tease short people for being slow.

File: code/modules/mob/living/carbon/human/species_types/psyphoza.dm [hunk 1]
```diff
@@ -32,6 +32,8 @@
 	//Fire bad!
 	burnmod = 1.25
 
+	species_height = SPECIES_HEIGHTS(8, 16, 0)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species_types/psyphoza.dm]: Is this right? All of the others have tiny values

File: code/__DEFINES/DNA.dm [hunk 2]
```diff
@@ -179,3 +179,6 @@
 #define G_MALE 1
 #define G_FEMALE 2
 #define G_PLURAL 3
+
+//Size / height stuff
+#define SPECIES_HEIGHTS(x, y, z) list("Normal" = x, "Short" = y, "Tall" = z)
```
Comment (PowerfulBacon) [inline code/__DEFINES/DNA.dm]: The ordering of normal, short, tall doesn't make much sense. Short, normal, tall would be more sensible.

**Comment by Tsar-Salat** [comment]
> This helps us represent species that are typically shorter than humans, but have at-least one height setting that is 'normal'.

What metric did you use to determine this?

File: code/__DEFINES/DNA.dm (hunk 2 above)
Comment (DrDuckedGoose) [reply code/__DEFINES/DNA.dm]: Fair enough

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**